### PR TITLE
update assignment - next.js

### PR DESCRIPTION
**Pull request by mohammedmulazada**
This PR will update the FE assignment as the old one was outdated and would not work out of the box for most candidates:

* Add next.js which is what the company is moving towards
* Add react-testing-library
* Update storybook
* Use typescript

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,42 +1,40 @@
 {
-  "name": "frontend-dev-assignment",
+  "name": "frontend-tech-assignment",
```
Comment (laura-debijenkorf) [inline package.json:2]: 🤣

File: src/pages/__tests__/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { render, screen } from '@testing-library/react'
```
Comment (laura-debijenkorf) [inline src/pages/__tests__/index.test.jsx:1]: no ending with ; ?

File: src/stories/Introduction.mdx [hunk 3]
```diff
@@ -0,0 +1,230 @@
+import { Meta } from '@storybook/blocks';
```
Comment (laura-debijenkorf) [inline src/stories/Introduction.mdx:1]: was this created automatically?

**Review by laura-debijenkorf — COMMENTED** [review]
There are a bunch of random images around...I believe they were added automatically..can you remove them?

File: src/stories/Introduction.mdx (hunk 3 above)
Comment (mohammedmulazada) [reply src/stories/Introduction.mdx:1]: Yeah this one was created automatically :) I'm happy to remove it!

File: src/pages/__tests__/index.test.jsx (hunk 2 above)
Comment (mohammedmulazada) [reply src/pages/__tests__/index.test.jsx:1]: Will add.

File: src/pages/api/search.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { suggestions } from "@/data/searchSuggestions";
```
Comment (laura-debijenkorf) [inline src/pages/api/search.ts:1]: Correct me if I'm wrong, but I don't remember this being provided in the previous assignment...

Comment (mohammedmulazada) [reply src/pages/api/search.ts:1]: It used to be (you can see it here https://github.com/deBijenkorf/frontend-assignment/blob/master/api/_search.get.json), or did you maybe mean something else?

Comment (laura-debijenkorf) [reply src/pages/api/search.ts:1]: Sorry, I meant the entire file 😅

**Review by corradin — DISMISSED** [review]

**Review by laura-debijenkorf — APPROVED** [review]

**Pull request merged**